autoscheduler/frontend/src/utils/throttle.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * Function to throttles the execution of other functions
+ * @param id ID to use, each ID operates on a separate timer
+ * @param callback Callback to run after delay expires
+ * @param delay Delay (in ms) before running effect
+ * @param runOtherIDs Whether to immediately run pending callbacks for IDs other than specified id
+ */
+const throttle = ((): (id: string, callback: any, delay: number, runOtherIDs: boolean) => void => {
```

Comment:
Do you think this function deserves tests of its own? It does have a decent amount of branching/logic to it

Reply:
I tested all of the functionality in the `saves course cards` tests for `CourseSelectColumn` except for the timer (since having any tests using a timer even in a different file was causing a lot of other tests to fail), I can see why we'd want to have tests for it to be more thorough but I feel like as long as the behavior from our components is correct it's fine. I can add them if you want though